### PR TITLE
Fix the ExecuteMsgFns macro in case there is a where clause

### PR DESCRIPTION
**Pull request by Kayanski**
This PR aims at fixing the macro when you are creating the ExecuteMsgFns macro along with a generic type and a where clause

### CODE REVIEW

**Review by CyberHoward — CHANGES_REQUESTED** [review]
Could you add a test-case?

**Comment by Kayanski** [comment]
T

> Could you add a test-case?

Added tests.
Test are failing because we need to update osmosis-std to fix them. It's already fixed in  : https://github.com/AbstractSDK/cw-orchestrator/pull/152

File: packages/cw-orch-fns-derive/src/execute_fns.rs [hunk 1]
```diff
@@ -75,16 +75,29 @@ pub fn execute_fns_derive(input: DeriveInput) -> TokenStream {
     });
 
     let derived_trait = quote!(
-        pub trait #bname<Chain: ::cw_orch::prelude::CwEnv, #type_generics>: ::cw_orch::prelude::CwOrchExecute<Chain, ExecuteMsg = #entrypoint_msg_type #ty_generics #where_clause> {
+        pub trait #bname<Chain: ::cw_orch::prelude::CwEnv, #type_generics>: ::cw_orch::prelude::CwOrchExecute<Chain, ExecuteMsg = #entrypoint_msg_type #ty_generics> #where_clause {
```
Comment (Buckram123) [inline packages/cw-orch-fns-derive/src/execute_fns.rs:78]: :exploding_head:

File: packages/cw-orch-fns-derive/src/execute_fns.rs [hunk 2]
```diff
@@ -75,16 +75,29 @@ pub fn execute_fns_derive(input: DeriveInput) -> TokenStream {
...
+        .map(|w| {
+            quote!(
+                #w #necessary_where
+            )
+        })
+        .unwrap_or(quote!(
+            where
+                #necessary_where
+        ));
+
```
Comment (Buckram123) [inline packages/cw-orch-fns-derive/src/execute_fns.rs:96]: Nice approach

**Review by Buckram123 — APPROVED** [review]
That's pretty cool

**Comment by CyberHoward** [comment]
> T
> 
> > Could you add a test-case?
> 
> Added tests. Test are failing because of the way osmosis handles dependencies and versions. There are some misdependencies between osmosis test tube packages

Could we get the tests resolved before merging this?

**Comment by CyberHoward** [comment]
Created an issue on test-tube: https://github.com/osmosis-labs/test-tube/issues/38

**Review by CyberHoward — APPROVED** [review]

**Pull request merged**